include/deal.II/lac/precondition.h

```diff
@@ -811,19 +811,54 @@ class PreconditionPSOR : public PreconditionRelaxation<MatrixType>
+ * eigenvalue down to a given lower eigenvalue specified by the optional
+ * parameter @p smoothing_range. The typical use case for the preconditioner
+ * is a Jacobi preconditioner specified through DiagonalMatrix, in which case
+ * the Chebyshev iteration corresponds to a Jacobi preconditioner if the
+ * degree variable is set to zero.
  *
- * This class is useful e.g. in multigrid smoother objects, since it is
- * trivially %parallel (assuming that matrix-vector products are %parallel).
+ * Apart from the inner preconditioner object, this iteration does not need
+ * access to matrix entries, which make it an ideal ingredient for matrix-free
+ * computations. In that context, this class can be use in multigrid smoother
+ * objects, since it is trivially %parallel (assuming that matrix-vector
+ * products are %parallel and the inner preconditioner is %parallel). Its use
+ * is demonstrated in the step-37 tutorial program.
  *
- * @author Martin Kronbichler, 2009; extension for full compatibility with
+ * <h4>Algorithm execution</h4>
+ *
+ * The Chebyshev method relies on an estimate of the eigenvalues of the matrix
+ * which are computed during the first invocation of vmult(). The algorithm
+ * invokes a conjugate gradient solver so symmetry and positive definiteness
+ * of the (preconditioned) matrix system is a strong requirement. This
+ * computation needs to be deferred until the first vmult() because the class
+ * needs to initialize temporary vectors in the same layout as source and
+ * destination vectors that are only available through vmult().
+ *
+ * The estimation of eigenvalues can also be bypassed by setting
+ * PreconditionChebyshev::AdditionalData::eig_cg_n_iterations to zero and
+ * providing sensible values for the largest eigenvalues in the field
+ * PreconditionChebyshev::AdditionalData::max_eigenvalue. If the range
+ * <tt>[max_eigenvalue/smoothing_range, max_eigenvalue]</tt> contains all
+ * eigenvalues of the matrix and the iteration number is high enough, this
+ * class can also be used as a direct solver.
+ *
+ * <h4>Requirements on templated classes</h4>
+ *
+ * For use with this class, the class MatrixType must be derived from
+ * Subscriptor because a SmartPointer to MatrixType is held in the class. In
+ * particular, this means that the matrix object needs to persist during the
+ * lifetime of PreconditionChebyshev. Regarding the preconditioner class, the
+ * object needs to be default-constructible and copy-constructible. The
+ * variable is held in a variable of type AdditionalData that is kept inside
+ * the class.
+ *
+ * @author Martin Kronbichler, 2009, 2016; extension for full compatibility with
  * LinearOperator class: Jean-Paul Pelteret, 2015
  */
-template <typename MatrixType=SparseMatrix<double>, class VectorType=Vector<double> >
+template <typename MatrixType=SparseMatrix<double>, typename VectorType=Vector<double>, typename PreconditionerType=DiagonalMatrix<VectorType> >
```

Comment:
looking through the code, i see why you called it `PreconditionerType` as you need `vmult` of that class. How about `DiagonalOperator`?